### PR TITLE
Enable memory_2d map for craftassist agent

### PR DESCRIPTION
**Pull request by aqibisbiqa**
# Description

The map that displays in the in the memory_2d pane for the Locobot agent was not previously implemented for the Craftassist agent. With this PR, we can now see the agent, obstacles from the obstacle map, and other detected objects as points plotted on the map pane.

Fixes # (issue)

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

Before:

Screen Shot 2022-05-20 at 4 26 50 PM

After:

Screen Shot 2022-06-03 at 2 34 22 PM

# Testing

You can see a map :) And the agent program + dashboard doesn't break when rendering the map after a multitude of test commands. This diff won't affect any of the previously implemented unit tests, and it seems to pass all of the existing ones.

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [x] I have added Docstrings and comments to the code.
- [x] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [x] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.

### CODE REVIEW

**Comment by ys1382fb** [comment]
You should be able to check some of the items in the Checklist, such as around testing and documentation.

File: agents/craftassist/craftassist_agent.py [hunk 1]
```diff
@@ -290,6 +290,19 @@ def perceive(self, force=False):
             self.memory.update(sem_seg_perception_output)
         self.areas_to_perceive = []
         self.update_dashboard_world()
+        # 5. self location
+        # FIXME better pose object
+        ###perception_output = perception_output._replace(self_pose=(x, z, yaw))
```
Comment (kavyasrinet) [inline agents/craftassist/craftassist_agent.py:295]: nit : remove comments

File: agents/craftassist/craftassist_agent.py [hunk 2]
```diff
@@ -290,6 +290,19 @@ def perceive(self, force=False):
...
+        #perception_output = perception_output._replace(self_pose=(0, 0, 0))
+        
+        if self.opts.draw_map == "memory":
+            # draw the map from memory
+            self.draw_map_to_dashboard()
+        elif self.opts.draw_map == "observations": # else draw directly from current obs
+            ###self.draw_map_to_dashboard(obstacles={}, xyyaw=(x,z,yaw))
+            self.draw_map_to_dashboard(obstacles={}, xyyaw=(0,0,0))
+        else:
```
Comment (kavyasrinet) [inline agents/craftassist/craftassist_agent.py:304]: we can skip this

**Review by kavyasrinet — COMMENTED** [review]
Any idea why the rendered screenshot in the "After" section only shows a snippet of the memory grid and doesn't render it fully in the pane ?

File: agents/craftassist/craftassist_agent.py (hunk 2 above)
Comment (kavyasrinet) [reply agents/craftassist/craftassist_agent.py:304]: Ah yes, good catch. Let's do that

**Pull request closed without merging**